runtime/src/benchmarking/criterion.rs

```diff
@@ -0,0 +1,147 @@
+/// Criterion benchmark executor implementations.
+use criterion::async_executor::AsyncExecutor;
+use futures::Future;
+use std::any::Any;
+use std::cell::RefCell;
+use std::thread_local;
+
+thread_local! {
+    static CONTEXT: RefCell<Option<Box<dyn Any + Send>>> = RefCell::new(None);
+}
+
+/// Set the context value
+fn set_context<C: Clone + Send + 'static>(context: C) {
+    CONTEXT.with(|cell| {
+        *cell.borrow_mut() = Some(Box::new(context));
+    });
+}
+
+/// Get the context value
+pub fn context<C: Clone + Send + 'static>() -> C {
```

Comment:
Instead of cloning, we should take the context. This is much more aligned with how it should be used.